tst/EnergyPlus/unit/ZoneEquipmentManager.unit.cc

```diff
@@ -5268,3 +5268,63 @@ TEST_F(EnergyPlusFixture, SpaceHVACMixerTest)
     EXPECT_NEAR(mixSpace2Node.MassFlowRateMaxAvail, equipInletNode.MassFlowRateMaxAvail * mixSpace2.fraction, 0.0001);
     EXPECT_NEAR(mixSpace3Node.MassFlowRateMaxAvail, equipInletNode.MassFlowRateMaxAvail * mixSpace3.fraction, 0.0001);
 }
+
+TEST_F(EnergyPlusFixture, ZoneEquipmentManager_GetZoneEquipmentTest)
+{
+
+    std::string const idf_objects = delimited_string({
+        "Zone,",
+        "  Space;                   !- Name",
+
+        "ZoneHVAC:EquipmentConnections,",
+        " Space,                    !- Zone Name",
+        " Space Equipment,          !- Zone Conditioning Equipment List Name",
+        " Space In Node,            !- Zone Air Inlet Node or NodeList Name",
+        " Space Exh Nodes,          !- Zone Air Exhaust Node or NodeList Name",
+        " Space Node,               !- Zone Air Node Name",
+        " Space Ret Node;           !- Zone Return Air Node Name",
+
+        "ZoneHVAC:EquipmentList,",
+        " Space Equipment,          !- Name",
+        " SequentialLoad,           !- Load Distribution Scheme",
+        " Fan:ZoneExhaust,          !- Zone Equipment 1 Object Type",
+        " Exhaust Fan,              !- Zone Equipment 1 Name",
+        " 1,                        !- Zone Equipment 1 Cooling Sequence",
+        " 1,                        !- Zone Equipment 1 Heating or No - Load Sequence",
+        " ,                         !- Zone Equipment 1 Sequential Cooling Fraction",
+        " ;                         !- Zone Equipment 1 Sequential Heating or No-Load Fraction",
+
+        "Fan:ZoneExhaust,",
+        "Exhaust Fan,               !- Name",
+        ",                          !- Availability Schedule Name",
+        "0.338,                     !- Fan Total Efficiency",
+        "125.0000,                  !- Pressure Rise{Pa}",
+        "0.3000,                    !- Maximum Flow Rate{m3/s}",
+        "Exhaust Fan Inlet Node,    !- Air Inlet Node Name",
+        "Exhaust Fan Outlet Node,   !- Air Outlet Node Name",
+        "Zone Exhaust Fans;         !- End - Use Subcategory",
+
+        "NodeList,",
+        "  Space Exh Nodes,  !- Name",
+        "  Space ZoneHVAC Exh Node, !- Node 1 Name",
+        "  Exhaust Fan Inlet Node; !- Node 1 Name",
+
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+    EXPECT_FALSE(has_err_output());
+    bool ErrorsFound = false;
+
+    // Test 0: The get flag should default to true
+    EXPECT_TRUE(state->dataZoneEquipmentManager->GetZoneEquipmentInputFlag);
+
+    // Test 1: This should return no errors and the get flag should now be false.
+    GetZoneData(*state, ErrorsFound);
+    AllocateHeatBalArrays(*state);
+    GetZoneEquipment(*state);
+    EXPECT_FALSE(ErrorsFound);
+    EXPECT_FALSE(state->dataZoneEquipmentManager->GetZoneEquipmentInputFlag);
```

Comment:
1) is this unit test still applicable to the latest code change?, and 2) I think it would be better to verify what arrays are allocated/filled with the data expected to be available to the simulation at the point where everything is now read in.

e.g.:

    EXPECT_GT(int(state.dataZoneEquip->ZoneEquipConfig.size()), 0); 
    EXPECT_EQ(state.dataZoneEquip->ZoneEquipConfig(1).ZoneNode, 1);

same for equipment list and whatever else is now available. This would still be a weak unit test but at least it would lock in what the current early call does.

Reply:
Well, it was just a suggestion. What you did with getting the zone inputs early is more important to me than this unit test. Not all code changes require a unit test.

Reply:
Ok...added some stuff here as requested.  It's still pretty basic, but this is better at verifying that it really isn't going through the actual code of GetZoneEquipment a second time.  Better?